include/cytnx_error.hpp

```diff
@@ -31,6 +33,16 @@ static inline void error_msg(char const *const func, const char *const file, int
             file, line);
     va_end(args);
     // std::cerr << output_str << std::endl;
+    std::cerr << output_str << std::endl;
+    std::cerr << "Stack trace:" << std::endl;
+    void *array[10];
+    size_t size;
+    size = backtrace(array, 10);
+    char **strings = backtrace_symbols(array, size);
+    for (size_t i = 0; i < size; i++) {
+      std::cerr << strings[i] << std::endl;
+    }
+    free(strings);
```

Comment:
Without it, it’s hard to know what cause the error

Reply:
Does setting `-Og` flag for the debug build and using the debugger help? `execinfo.h` is linux specific, do we have to enable it when on the Linux and building with the debug mode?

Reply:
When using debugger it's kinda hard to know where to set the breakpoint, since when run in python, the error won't terminate the process iirc.

Also, by using debug build, you'll need to set excess flags to be able to run with -fsanitize=address and cuda.(Written in Install.sh)

Reply:
For macos installation, I should add a macro guard

Reply:
Also, it's kinda weird for me that, an error occurred without stacktrace. Not sure if that's convention.

Reply:
> Also, by using debug build, you'll need to set excess flags to be able to run with -fsanitize=address and cuda.(Written in Install.sh)

You are right. We need to set some environment variables when executing Python code binding with the debug build. However, only the debug build includes enough information, like symbol names, and line positions, for debugging. For me, I always write the minimal failing code in C++ to use the debugger. I don't know if there is a better practice.

> Also, it's kinda weird for me that, an error occurred without stacktrace. Not sure if that's convention.

I am unsure how other Python libraries handle their errors thrown from the executable files, even on Windows, either.

Reply:
I agree that it is useful in debug mode, but I have doubts about the stack trace providing valuable information in the release build, as it optimizes away many details. However, printing the stack trace may not significantly impact performance since we expect that the program will crash after displaying the error message. Therefore, it could also be fine  to include this in the release build.

Reply:
If you feel it is necessary for the release build, just resolve this thread after ensuring that execinfo.h is included only for Linux.

Reply:
Can we make an int on stack so that user can change the value in runtime? Basically something like python logging